astropy/time/core.py

```diff
@@ -2787,32 +2800,6 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
         result._location = location
         return result
 
-    def __array_function__(self, function, types, args, kwargs):
```

Comment:
Move to `TimeBase` so `TimeDelta` benefits too. In principle, they might need different overrides, but so far there seems to be little reason to split them.